src/service-module/types.ts

```diff
@@ -59,3 +64,268 @@ export interface MakeServicePluginOptions {
+   * this method like MyModel.merge(this, newData).
+   * @param dest destination object
+   * @param source source object
+   * @param blacklist keys to ignore when merging
+   * @example
+   * class Todo extends BaseModel {
+   *   public constructor(data, options?) {
+   *   options.merge = false // Prevent the internal merge
+   *   super(data, options)
+   *   // ... your custom constructor logic happens here.
+   *   // Call the static merge method to do your own merging.
+   *   Todo.merge(this, data)
+   *   }
+   * }
+   */
+  merge(dest: unknown, source: unknown, blacklist?: string[]): void
+
+  /**
+   * Create new Model
+   * @param data partial model data
+   * @param options model instance options
+   */
+  new (data?: AnyData, options?: ModelInstanceOptions): Model
+  prototype: Model
+
+  /**
+   * The instanceDefaults API was created in version 1.7 to prevent
+   * requiring to specify data for new instances created throughout
+   * the app. Depending on the complexity of the service's "business
+   * logic", it can save a lot of boilerplate. Notice that it is
+   * similar to the setupInstance method added in 2.0. The instanceDefaults
+   * method should ONLY be used to return default values for a new
+   * instance. Use setupInstance to handle other transformations on
+   * the data.
+   * @param data the instance data
+   * @param ctx setup context
+   */
+  instanceDefaults(data: AnyData, ctx: ModelSetupContext): AnyData
+
+  /**
+   * A new setupInstance class method is now available in version 2.0.
+   * This method allows you to transform the data and setup the final
+   * instance based on incoming data. For example, you can access the
+   * models object to reference other service Model classes and create
+   * data associations.
+   * @param data the instance data
+   * @param ctx setup context
+   */
+  setupInstance(data: AnyData, ctx: ModelSetupContext): AnyData
+
+  /**
+   * Gets called just before sending the data to the API server. It gets
+   * called with the data and must return the diffed data.
+   *
+   * Default: `data => data`
+   * @param data the instance data
+   */
+  diffOnPatch(data: AnyData): AnyData
+
+  /**
+   * A proxy for the `find` action
+   * @param params Find params
+   */
+  find<M extends Model = Model>(params?: Params): Promise<M[] | Paginated<M>>
+  /**
+   * A proxy for the `find` getter
+   * @param params Find params
+   */
+  findInStore<M extends Model = Model>(params?: Params): Paginated<M>
+
+  /**
+   * A proxy for the `get` action
+   * @param id ID of record to retrieve
+   * @param params Get params
+   */
+  get<M extends Model = Model>(id: Id, params?: Params): Promise<M | undefined>
+  /**
+   * A proxy for the `get` getter
+   * @param id ID of record to retrieve
+   * @param params Get params
+   */
+  getFromStore<M extends Model = Model>(id: Id, params?: Params): M | undefined
+}
+
+/** Model instance interface */
+export interface Model {
+  [key: string]: any
```

Comment:
This allows indexing Models by any string and the value will have type `any`.

Ideally I would get rid of this line and force people to define an interface for the model's underlying data (as I outlined in the new docs), but that would make this a breaking change so probably best to keep this and make stricter-typing opt-in instead of breaking+opt-out.